src/System.Private.CoreLib/src/System/Globalization/CompareInfo.Unix.cs

```diff
@@ -175,11 +175,6 @@ private unsafe int IndexOfCore(string source, string target, int startIndex, int
                 return IndexOfOrdinal(source, target, startIndex, count, ignoreCase: false);
             }
 
-            if (_isAsciiEqualityOrdinal && CanUseAsciiOrdinalForOptions(options) && source.IsFastSort() && target.IsFastSort())
```

Comment:
The IsFastSort, etc. bits are not cached in object pre-header in CoreRT. I agree we should keep the CoreCLR specific code under `#if CORECLR` so that we can share the sources.